SharkCage/CageManager/CageManager.cpp

```diff
@@ -300,24 +371,93 @@ bool CreateACL(std::unique_ptr<PSID, decltype(local_free_deleter<PSID>)> group_s
 	PROCESS_INFORMATION process_info = {};
 	std::vector<wchar_t> path_buf(path.begin(), path.end());
 	path_buf.push_back(0);
-	if (::CreateProcess(NULL, path_buf.data(), NULL, NULL, TRUE, 0, NULL, NULL, &info, &process_info))
+
+	if (::CreateProcess(NULL, path_buf.data(), NULL, NULL, TRUE, 0, NULL, NULL, &info, &process_info) != 0)
 	{
 		std::cout << "Failed to start process. Err " << ::GetLastError() << std::endl;
 	}
 
+	std::optional<PROCESS_INFORMATION> process_info_additional_app;
+	if (additional_app.has_value() && additional_app_path.has_value())
+	{
+			std::vector<wchar_t> path_buf_keepass(additional_app_path.value().begin(), additional_app_path.value().end());
+			path_buf_keepass.push_back(0);
+			process_info_additional_app = { 0 };
+			STARTUPINFO info_additional_app = { 0 };
+			info_additional_app.lpDesktop = new_desktop_name_buf.data();
+
+			if (::CreateProcess(NULL, path_buf_keepass.data(), NULL, NULL, TRUE, 0, NULL, NULL, &info_additional_app, &process_info_additional_app.value()))
+			{
+				std::cout << "Failed to start process in lalalalal. Err " << GetLastError() << std::endl;
+			}
+	}
+
+	// Start a thread that will perform the main task of the service
+	//worker_thread = ::CreateThread(NULL, 0, ManagerWorkerThread, NULL, 0, NULL);
+	//::WaitForSingleObject(worker_thread, INFINITE);
+
 	desktop_thread.join();
+	receiving_messages = false;
 
 	//::SendMessage(nullptr /*main handle of new process*/, WM_CLOSE, NULL);
 	// with timeout, if nothing happens, terminate process
+	if (process_info_additional_app.has_value() && !::TerminateProcess(process_info_additional_app.value().hProcess, 0))
+	{
+		std::cout << "Failed to terminate process Err " << ::GetLastError() << std::endl;
+	}
+
 	if (!::TerminateProcess(process_info.hProcess, 0))
 	{
 		std::cout << "Failed to terminate process Err " << ::GetLastError() << std::endl;
 	}
 
 	// wait for the process to exit
 	::WaitForSingleObject(process_info.hProcess, INFINITE);
```

Comment:
do we have to wait for the process if you terminate it above? why terminate it in the first place?